transformations/city_names_transformation/transformation.py

```diff
@@ -0,0 +1,256 @@
+                    ind += 1
+            spans.append({"Word": span, "Entity": ent})
+        return spans
+
+    def transform(self, doc, seed=None):
+        """
+        Given a SpaCy doc object, this returns a sentence string that replaces populous cities with less populous/well-known cities
+
+        Parameters
+        ----------
+        doc : SpaCy Doc
+            SpaCy doc object that is ouputted from a SpaCy model given an input sentence.
+        seed : int, optional
+            seed value for random operations. The default is None.
+
+        Returns
+        -------
+        new_sentence : str
+            The transformed sentence.
+
+        """
+        if seed is not None:
+            random.seed(seed)
+        ents_dict = self.create_ents_dict(doc)
+        sent_words = []
+        for i in ents_dict:
+            if i["Word"] in list(self.populous_cities) and (
+                i["Entity"] == "GPE" or i["Entity"] == "LOC"
+            ):
+                sent_words.append("<CITY>")
+            else:
+                sent_words.append(i["Word"])
+        new_sentence = " ".join(sent_words)
+        while "<CITY>" in new_sentence:
+            rand_city = self.scarce_cities[
+                random.randint(0, len(self.scarce_cities))
+            ]
+            new_sentence = new_sentence.replace("<CITY>", rand_city, 1)
+        return new_sentence
+
+
+"""
+CityNamesTransformation is a class that inherits from the SentenceOperation interface.
+This class's main purpose is to take an input sentence that contains a populous city and replace the instance of that city with a new city that is
+not as populous or well-known to test the robustness of NLP models.
+"""
+
+
+class CityNamesTransformation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TAGGING
+    ]
+    languages = ["en", "es"]
+    keywords = ["lexical","model-based","tokenizer-required","highly-meaning-preserving","high-precision","low-coverage","high-generations","world-knowledge"]
+    # languages the operation can operate on.
+    def __init__(self, seed=0, max_outputs=1, lang="en", data_path=None):
+        """
+        Constructor of the CityNamesTransformation object in a given language
+
+        Parameters
+        ----------
+        seed : int, optional
+            seed value for random operations. The default is 0.
+        max_outputs : 1, optional
+            How many ouput sentences can be created by an operation. The default is 1.
+        lang : str, optional
+            What language you want to perform the operation on. The default is "en".
+        data_path : str, optional
+            data path of database of English and Spanish cities. The default is None.
+
+        Returns
+        -------
+        None.
+
+        """
+        super().__init__(seed, max_outputs=max_outputs)
+        if lang == "en":
+            self.model = spacy.load("en_core_web_sm")
```

Comment:
You might want to use spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/d39f0e5f57a79937296351996617ac03ee80fb74/transformations/close_homophones_swap/transformation.py#L55).